File: jifrozen/구현/문자열_압축.java
public class 문자열_압축 {
    public int solution(String s) {
        int result = s.length();
        String compression = "";
        String answer = "";
        for (int i = 1; i < s.length() / 2 + 1; i++) {
            compression = s.substring(0, i);
            int cnt = 1;
            for (int j = i; j < s.length(); j = j + i) {
                String sub = "";
                for (int k = j; k < j + i; k++) {
                    if (k < s.length()) {
                        sub += s.charAt(k);
                    }
                }
                if (compression.equals(sub)) {
                    cnt += 1;
                } else {
                    if (cnt < 2) {
                        answer += compression;
                    } else {
                        answer += cnt + compression;
                    }
                    compression = sub;
                    cnt = 1;
                }
            }
            if (cnt < 2) {
                answer += compression;
            } else {
                answer += cnt + compression;
            }
            result = Math.min(result, answer.length());
            answer = "";
        }
        return result;
    }

}

File: yunva17/구현/문자열_압축.py
s = input()

def solution(s):
    answer = 0
    return answer

result = len(s)

for i in range(1, len(s) //2 +1):
    cut = s[0:i]
    string=""
    count = 1
    for j in range(i, len(s)+1, i):
        if(cut == s[j:j+i]):
            count+=1
        else:
            if count!=1:
                string += str(count)+cut
                cut = s[j:j+i]
                count = 1
            else:
                string += cut
                cut = s[j:j+i]
    if count!=1:
        string += str(count)+cut
    else:
        string += cut
    result = min(len(string), result)

print(result)
